src/controllers/category.js

```diff
@@ -1,11 +1,21 @@
 const categoryService = require('~/services/category')
 
+const getCategories = async (req, res) => {
+  const skip = isNaN(parseInt(req.query.skip)) ? undefined : parseInt(req.query.skip)
+  const limit = isNaN(parseInt(req.query.limit)) ? undefined : parseInt(req.query.limit)
```

Comment:
The ternary operator in this case is functional but could be simplified with a helper function like this
onst parseQueryInt = (value, defaultValue) => {
  const parsedValue = parseInt(value);
  return isNaN(parsedValue) ? defaultValue : parsedValue;
};
and then
const skip = parseQueryInt(req.query.skip, 0);
...